standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreTServerSocket.java

```diff
@@ -29,8 +32,32 @@
  * accepted sockets.
  *
  */
-public class TServerSocketKeepAlive extends TServerSocket {
-  public TServerSocketKeepAlive(TServerSocket serverSocket) throws TTransportException {
+public class MetastoreTServerSocket extends TServerSocket {
+  private final Configuration configuration;
+
+  public MetastoreTServerSocket(TServerSocket serverSocket,
+      Configuration conf) throws TTransportException {
     super(serverSocket.getServerSocket());
+    this.configuration = Objects.requireNonNull(conf);
+  }
+
+  @Override
+  public TSocket accept() throws TTransportException {
+    TSocket ts = super.accept();
+    boolean keepAlive = MetastoreConf.getBoolVar(configuration, MetastoreConf.ConfVars.TCP_KEEP_ALIVE);
+    if (keepAlive) {
```

Comment:
it doesn't seen equivalent to original code. I don't see explicit set of setKeepAlive on socket when keepAlive==true

Reply:
the check is performed outside of the class before the PR, https://github.com/apache/hive/pull/5699/files#diff-00e70b6958060aa36762b21bf16676f83af01c1e09b56816aecc6abe7c8ac866L634

Reply:
yes, but were. did we execute `setKeepAlive` before?

Reply:
it's introduced in HIVE-1410 and removed by HIVE-25098(not sure why we remove it on upgrading the thrift lib)

Reply:
nice :)
but do we need the `if check`? can't we simply do `ts.getSocket().setKeepAlive(keepAlive);`

Reply:
done

Reply:
After some check, the default `keepAlive` in thrift 0.16 is true, https://github.com/apache/thrift/blob/0.16.0/lib/java/src/org/apache/thrift/transport/TSocket.java#L78, so `KeepAlive` is always true whether `hive.metastore.server.tcp.keepalive` is true or not based on [HIVE-1410](https://issues.apache.org/jira/browse/HIVE-1410)

Since the `KeepAlive` helps HMS prevent the half-open connections, so it would be good to enable it, so I would like to keep the [HIVE-25098](https://issues.apache.org/jira/browse/HIVE-25098).

Reply:
why don't we drop the config then?

Reply:
this will remove the property in `metaVars`, I would like to leave it in a separate jira, to make the fix port to old releases easily

Reply:
ok